src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs

```diff
@@ -102,7 +102,7 @@ private void ProcessToken(JToken token)
                     case JTokenType.Bytes:
                     case JTokenType.TimeSpan:
                         string key = ConfigurationSectionNames.JobHost + ConfigurationPath.KeyDelimiter + ConfigurationPath.Combine(_path.Reverse());
-                        Data[key] = token.Value<JValue>().ToString(CultureInfo.InvariantCulture);
+                        Data[key] = token.Value<string>()?.ToString(CultureInfo.InvariantCulture);
```

Comment:
The right fix was in the earlier PR. We do not want parse other token types as string. This change makes sense for token string.

Reply:
Existing code is converting the token to string using ToString(). Difference is that previous change converts functionTimeout: null to string.empty while this keeps it null

Reply:
JValue is the right type we need to convert to. Null is a right type. The issue is in the way code was parsing the fucntiontimeout specifically